### PR TITLE
feat(events)!: membership eligibility Phase 2 — gate stack on the paid path (#831)

### LINKED ISSUE #831: feat: membership eligibility Phase 2 — run the gate stack on the paid path (unblock gated + paid tiers)
## Problem

**A paid membership tier cannot be gated by a questionnaire or by manual approval.** This is a
product-level blocker: an organization that vets applicants *and* charges dues has no way to
model that, which is the normal shape of a real membership club.

Two mirrored guards enforce the exclusion today:

- `events/service/subscription_service.py:74` — `_assert_tier_ungated()`: 400 when putting an
  active plan on a tier that sets `requires_membership_approval` or `membership_questionnaire`.
- `events/service/organization_service/membership.py:494` — `_assert_tier_not_monetized()`:
  400 when enabling either gate on a tier that already has an active plan.

They are correct *given the current architecture*, and both carry `ponytail:` notes saying so.
The underlying reason is that there are two parallel join paths that do not talk to each other:

| path | entry point | runs the gate stack? |
| --- | --- | --- |
| free | `POST /me/organizations/{slug}/apply` | yes — all 10 gates |
| paid | `POST /me/organizations/{org_id}/subscribe` | **no** — straight to Stripe Checkout (`controllers/me_subscriptions.py:129`) |

So on a monetized tier the gate config would go silently inert, and the guards exist to stop an
organizer configuring something that quietly does nothing. `PaymentReadyGate` (gate #10) is
already stubbed for this: it rejects every plan-bearing application with `plan_not_online`, and
`/apply` refuses `plan_id` outright (`controllers/me_applications.py:100`).

## Proposal — Phase 2: one gated pipeline for both paths

Unify on the application state machine, with payment as the final step rather than a parallel
door.

1. **`/apply` accepts `plan_id`.** Drop the `plan_id` refusal in
   `MeMembershipApplicationsController.apply`. The application row already has `plan` and
   `subscription` FKs.
2. **`PaymentReadyGate` becomes real.** When every prior gate passes and the application carries
   a plan, it returns an *allowing* verdict with a new `next_step` (`checkout` / `requires_payment`)
   instead of blocking. When gates ahead of it block (questionnaire pending, approval pending),
   the application waits exactly as the free path does — the applicant fills the questionnaire,
   staff approve, and *then* the checkout step unlocks.
3. **`/subscribe` runs the gate stack.** Before `start_online_subscription`, build a
   `MembershipEligibilityService(user, organization, tier=plan.tier, plan=plan)` and refuse a
   non-allowed verdict with the serialized eligibility payload — the same 400 shape `/apply`
   already returns, which the frontend knows how to render.
4. **Checkout is reachable from an approved application.** Either `/subscribe` accepts an
   `application_id`, or the application detail exposes a `checkout_url`. Decide during
   implementation; the constraint is that an approved-but-unpaid application must have exactly
   one way forward.
5. **Membership is granted on payment, not on approval.** The Stripe activation webhook already
   creates the membership for the paid path; it must also settle the originating application
   (`status=completed`, link `subscription`).
6. **Delete both guards** and their tests once 1–5 land.

## Edge cases to settle during implementation

- **Approved, then never pays.** The application sits in an "awaiting payment" state. Needs an
  expiry, or it pollutes the staff board forever.
- **`TierAvailabilityGate` free-path block.** `Reasons.TIER_REQUIRES_SUBSCRIPTION` (gate #6,
  `gates.py:227`) blocks a tier-less/free apply on a monetized tier. That stays correct, but the
  verdict should now point the user at the plan rather than dead-ending.
- **Questionnaire re-evaluation between approval and payment.** `max_submission_age` could lapse
  in the gap. Decide whether checkout re-runs the gates (recommended: yes, at
  `start_online_subscription` time).
- **Plan switching on a gated tier.** `change-plan` and `revive` also bypass the gates today —
  the `ponytail:` comment on `_assert_tier_ungated` names all three. They need the same treatment
  or an explicit decision that an existing member is not re-gated.
- **Org-wide defaults.** The current guards deliberately only trip on *tier-level* overrides,
  because refusing on `default_membership_questionnaire` would block every paid tier in such an
  org. After Phase 2 the org defaults must apply to paid tiers too — that is a **behaviour change
  for existing orgs** that have both a default questionnaire and paid plans. Audit before
  deploying; those orgs' paid tiers become gated.

## Acceptance criteria

- [ ] A tier can simultaneously have an active plan and a membership questionnaire / approval
      requirement, and both are enforced.
- [ ] An applicant on such a tier: applies → fills the questionnaire → is approved → pays →
      becomes a member, with the application settling at each step.
- [ ] `/subscribe` refuses a gated-but-ineligible caller with the eligibility payload, and never
      opens a Checkout session for them.
- [ ] `_assert_tier_ungated` and `_assert_tier_not_monetized` are gone.
- [ ] `docs/architecture/membership-eligibility.md` updated: gate #10 diagram, the Phase 1 notes,
      and the "Phase 2" references throughout.
- [ ] Migration/audit note for orgs with a `default_membership_questionnaire` + paid plans.
- [ ] `make dump-openapi` run.

## Related

- Public tier listing (the other half of the deadlock — a gated tier is currently invisible).
- Free / non-renewing plans.

### PR DESCRIPTION
**Pull request by biagiodistefano**
Closes #831.

## What

Unifies the free and paid membership join paths on the application state machine, unblocking gated **and** monetized tiers (the production deadlock reported on the "revel events collective" org, together with #830).

- **`PaymentReadyGate` (gate #10) is real.** Allowing `PROCEED_TO_PAYMENT` verdict when the plan is payable; blocks on offline plan, org not Stripe-connected, paused/at-cap plan (`plan_unavailable`), duplicate non-terminal subscription — and on approval-required-with-no-application with the new `next_step=submit_application`. That last one matters: `ManualApprovalGate` deliberately falls through when nothing is on file (so `/apply` can create the row), and without the gate-10 backstop a cold `/subscribe` would have opened Checkout past staff approval.
- **`/apply` accepts `plan_id`.** Tier derived from the plan when `tier_id` is omitted; a re-POST with a different plan repoints the PENDING row; offline or cross-tier plans hard-block. Plan-bearing rows advance to `APPROVED` awaiting payment (machinery pre-built in #774).
- **`/subscribe` runs the full gate stack** and refuses with the same serialized eligibility 400 body `/apply` uses (response schema is now `MembershipEligibilitySchema | ErrorDetail`). One deliberate carve-out: a `duplicate_active_subscription` verdict falls through to the subscription machinery so the pending-checkout resume flow keeps returning its 409 + fresh `checkout_url`. The created subscription is linked to the caller's application.
- **Stripe activation settles the application**: `_settle_originating_application` runs inside `_ensure_active_member`, the funnel both webhook activation paths share — member ACTIVE and application COMPLETED in the same moment.
- **Approved-but-unpaid expiry**: new beat task `events.expire_stale_approved_applications` (daily 04:30 UTC, data migration `0108`) cancels plan-bearing APPROVED rows older than 30 days with no live linked subscription.
- **Both Phase-1 guards deleted** (`_assert_tier_ungated`, `_assert_tier_not_monetized`) with their messages, call sites, test file, and locale entries. `change-plan`/`revive` are deliberately **not** re-gated (existing/former members already passed the gates once) — documented in the architecture doc.
- `docs/architecture/membership-eligibility.md` rewritten for Phase 2 (gate-10 table, mermaid, next-step/reason tables, state machine, endpoints, coexistence section).

## Deploy note — org-default behaviour change (moot until first deploy)

**The membership/subscriptions stack (#774 and everything after) is not in the latest release (v1.72.1)**, so no production org can have plans yet and the audit below will return zero rows today. Kept for the record: once deployed,

org-wide defaults apply to paid tiers — any org combining `default_membership_questionnaire` or `default_requires_membership_approval=true` with active plans becomes gated on its paid tiers at deploy. Audit before deploying:

```sql
SELECT o.slug FROM events_organization o
WHERE (o.default_membership_questionnaire_id IS NOT NULL OR o.default_requires_membership_approval)
AND EXISTS (SELECT 1 FROM events_membershipsubscriptionplan p
            JOIN events_membershiptier t ON p.tier_id = t.id
            WHERE t.organization_id = o.id AND p.is_active);
```

## Test plan

- New: gate-10 readiness matrix (9 tests), paid `/apply` flows (6), gated `/subscribe` flows (4), application settlement (5), expiry sweep (5), coexistence (2), and a full end-to-end lifecycle test (`test_gated_paid_membership_flow.py`): cold subscribe blocked → apply → questionnaire → staff approve → checkout → activation → member ACTIVE + application COMPLETED → re-subscribe refused.
- ~750 tests green across every affected suite; `make check` (format, lint, mypy --strict, migration-check, i18n-check, file-length, task-names) passes; `make dump-openapi` run.
- Note for reviewers running scoped tests: combining many test files with the `test_organization_admin/` directory in one pytest invocation trips a pre-existing collection quirk (mass phantom "fixture not found" errors, reproducible on main) — run that directory in its own invocation.

Companion of #830 (public tier listing, merged) and #833 (questionnaire reverse relations, merged). Frontend counterpart: letsrevel/revel-frontend#720, E2E letsrevel/revel-frontend#723. #832 (free/non-renewing plans) lands on top of this.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Paid membership applications can now proceed through eligibility checks and payment checkout.
  * Membership tiers can combine paid plans with approval and questionnaire requirements.
  * Applications are linked to subscriptions and completed when payment activates membership.
  * Stale approved applications are automatically cancelled after 30 days.

* **Bug Fixes**
  * Added clearer handling for unavailable plans, duplicate subscriptions, missing approvals, and payment readiness.

### CODE REVIEW

File: src/events/tasks/subscriptions.py [hunk 1]
```diff
@@ -569,3 +569,48 @@ def resync_org_subscription_fees(org_id: str) -> "FeeResyncCounters":
...
+    cancelled = (
+        OrganizationMembershipRequest.objects.filter(
+            status=OrganizationMembershipRequest.Status.APPROVED,
+            plan__isnull=False,
+            updated_at__lt=cutoff,
+        )
+        # A row whose linked subscription is non-terminal is payment in flight.
+        # NULL subscriptions survive the exclude (NULL never matches __in).
+        .exclude(subscription__status__in=non_terminal)
+        .update(status=OrganizationMembershipRequest.Status.CANCELLED)
```
[the remark this answers is not in the document]
Comment (biagiodistefano) [reply src/events/tasks/subscriptions.py]: Fixed in 3c58c725 — the bulk update now stamps `updated_at=timezone.now()` and the sweep test asserts the cancellation timestamp is fresh.

**Comment by biagiodistefano** [comment]
### Code review

Found 2 issues (both fixed in 527ee5c5):

1. Cross-tier `change-plan` reopened the gate bypass this PR closes: `_validate_change_plan_target` never enforced same-tier, so a member could subscribe to an ungated tier's plan and self-service switch onto a gated tier's plan — the member-tier sync then repoints `member.tier` with no questionnaire/approval ever run. Structurally impossible under Phase 1 (gated tiers could not carry active plans), reopened by deleting the guards. Fixed by `ensure_tier_change_allowed` in the new `subscription_eligibility` service: cross-tier targets run the destination tier's gate stack; same-tier swaps stay un-re-gated.

https://github.com/letsrevel/revel-backend/blob/0e0e7b70ba25482c00e3b682def3374743497d73/src/events/service/subscription_service.py#L690-L719

2. The new `SUBMIT_APPLICATION` block serialized `reason="Your application is awaiting staff approval."` to a caller who by construction has no application on file — regressing the honesty rule the free path explicitly encodes (`check_eligibility`: "would be a lie when no application exists"). Fixed: the verdict now carries `reason_code` + `next_step` with no prose, mirroring the free path's shaping.

https://github.com/letsrevel/revel-backend/blob/0e0e7b70ba25482c00e3b682def3374743497d73/src/events/service/membership_manager/gates.py#L457-L463

Also applied below the reporting threshold: `/subscribe` orchestration extracted to the `subscription_eligibility` service (thin-controller policy), stale "Phase 2 `/pay`" comments and the `plan` FK `help_text` corrected (migration `0109` + locale catalogs), and `_ensure_active_member`'s docstring now documents the settlement side effect. Dismissed after verification: the paid-approval guard gap (downstream defenses hold: visibility 404, `create_subscription` BANNED/duplicate refusals, `_ensure_active_member` blocked/untouched branches) and the PAUSED-member settlement race (settling is the better trade-off — the alternative lets the sweep cancel a paid application).

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
